loadstring(game:HttpGet("https://raw.githubusercontent.com/noctryxzen/Getter/refs/heads/main/GetFS.luau"))()
loadstring(game:HttpGet("https://raw.githubusercontent.com/noctryxzen/NUI/refs/heads/main/Notification.luau"))()
if queue_on_teleport then
	queue_on_teleport([[loadstring(game:HttpGet("https://raw.githubusercontent.com/noctryxzen/Games/refs/heads/main/YouVSEveryone_FastestAutoFarm.luau"))()]])
else
	SendNotification({Title = "queue_on_teleport not supported", Text = "Please put the loadstring to your autoexec!", Duration = 15})
end
local TeleportPart = GetService.Workspace:WaitForChild("TeleportParts"):WaitForChild("TeleportPart1"):WaitForChild("TeleportPart1")
local Cash = GetService.Players.LocalPlayer:WaitForChild("leaderstats"):WaitForChild("Cash")
GetService.RunService.PreSimulation:Connect(function()
	firetouchinterest(GetService.Players.LocalPlayer.Character and GetService.Players.LocalPlayer.Character:WaitForChild("HumanoidRootPart"), TeleportPart, 0)
	firetouchinterest(GetService.Players.LocalPlayer.Character and GetService.Players.LocalPlayer.Character:WaitForChild("HumanoidRootPart"), TeleportPart, 1)
end)
Cash.Changed:Connect(function()
	task.wait(.1)
	GetService.TeleportService:Teleport(game.PlaceId, GetService.Players.LocalPlayer)
end)
task.delay(.2, function()
	GetService.TeleportService:Teleport(game.PlaceId, GetService.Players.LocalPlayer)
end)
